### PR TITLE
ARROW-1872: [Website] Minor edits and addition of YAML for versions

### PR DESCRIPTION
**Pull request by riboflavin**
Hi, in this version I've:

* Added a YAML file for tracking versions and outputting related links etc.
* Done some minor cleanup on HTML and images
* Added a small paragraph about what Arrow is to the front page

### CODE REVIEW

**Comment by wesm** [comment]
Cool, I'm updating this to reflect that it corresponds to https://issues.apache.org/jira/browse/ARROW-1872

File: site/index.html [hunk 1]
```diff
@@ -1,74 +1,69 @@
...
+    <h4><a href="{{ site.baseurl }}/blog/"><strong>See Latest News</strong></a></h4>
+    <div class="row">
         <div class="col-lg-4">
-          <h2>Fast</h2>
-          <p>Apache Arrow&#8482; enables execution engines to take advantage of
- the latest SIMD (Single input multiple data) operations included in modern
- processors, for native vectorized optimization of analytical data
- processing. Columnar layout is optimized for data locality for better
- performance on modern hardware like CPUs and GPUs.</p>
+            <p>Apache Arrow is an open source development platform for columnar in-memory data structures and processing. It defines a language-independent physical memory layout, enabling zero-copy, zero-deserialization interchange of flat and nested columnar data amongst a variety of systems such as Python, R, Apache Spark, ODBC protocols, and proprietary systems that utilize the open source components.</p>
```
Comment (wesm) [inline site/index.html]: We'll need to do a little bit of work on this, as this isn't the top-level pitch that was discussed most recently on the dev@ mailing list. 

Here was that pitch

> Apache Arrow is a cross-language development platform for in-memory data. It
specifies a standardized language-independent columnar memory format for flat
and hierarchical data, organized for efficient analytic operations on modern
hardware. It also provides computational libraries and zero-copy streaming
messaging and interprocess communication. Languages currently supported include
C, C++, Java, JavaScript, Python, and Ruby.

Comment (riboflavin) [reply site/index.html]: Got it. I've added a new commit with this language.

**Comment by wesm** [comment]
@riboflavin this visual layout doesn't look right to me. 

arrow_homepage

We might want to change the top "Powering Columnar In-Memory Analytics" tag line now

**Comment by riboflavin** [comment]
Absolutely. Hoping this looks better?

**Review by wesm — APPROVED** [review]
+1, formatting changes look good. Thanks @riboflavin!

**Comment by wesm** [comment]
@riboflavin can you let me know your JIRA id on https://issues.apache.org/jira so I can assign this issue to you?

**Comment by riboflavin** [comment]
@wesm try riboflavin?

**Comment by wesm** [comment]
done, thanks!

**Pull request closed without merging**